lk2nd/project/lk2nd.mk

```diff
@@ -11,11 +11,14 @@ ifneq ($(ENABLE_FBCON_DISPLAY_MSG),1)
 MODULES += $(if $(filter $(MODULES), lk2nd/display), lk2nd/device/menu)
 endif
 
-# Use part of the "boot" partition for the lk2nd boot image. The real Android
-# boot image can be placed in the partition with 512 KiB offset.
-LK2ND_PARTITION_BASE ?= boot
-LK2ND_PARTITION_NAME ?= lk2nd
-LK2ND_PARTITION_SIZE ?= 512*1024
+# Use part of the "boot" and "recovery" partition for the lk2nd boot image.
+# The real Android boot image can be placed in the partition with 512 KiB offset.
+LK2ND_BOOT_PARTITION_BASE ?= boot
```

Comment:
If the goal is making an image that would boot from recovery, I wonder if it's simpler to document how to compile and use an image with `make LK2ND_BOOT_PARTITION_BASE=recovery` instead of introducing complicated logic of checking and retrying images on multiple partitions that may or may not exist on the device...

Reply:
Discussed in chat: The mainline android + mainline recovery would both need to go via lk2nd and if android reboots into recovery, oem aboot will boot recovery partiiton, so lk2nd needs to be installed to both.